gidari.go

```diff
@@ -46,17 +47,37 @@ func NewConfig(ctx context.Context, file *os.File) (*Config, error) {
 }
 
 // TransportFile will construct the transport operation using a configuration YAML file.
-func TransportFile(ctx context.Context, file *os.File) error {
-	cfg, err := NewConfig(ctx, file)
+func TransportFile(ctx context.Context, file *os.File) (*Config, error) {
+	info, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get file stat for reading: %w", err)
+	}
+
+	bytes := make([]byte, info.Size())
+
+	_, err = file.Read(bytes)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file: %w", err)
+	}
+
+	cfg, err := transport.NewConfig(bytes)
 	if err != nil {
-		return fmt.Errorf("unable to create new config: %w", err)
+		return nil, fmt.Errorf("unable to create new config: %w", err)
+	}
+
+	// Disable logger
+	cfg.Logger.SetOutput(io.Discard)
+
+	return &Config{*cfg}, nil
+	if err != nil {
+		return nil, fmt.Errorf("unable to create new config: %w", err)
 	}
 
-	return Transport(ctx, cfg)
+	return nil, Transport(ctx, cfg)
 }
 
 // Transport will construct the transport operation using a "transport.Config" object.
-func Transport(ctx context.Context, cfg *Config) error {
+func Transport(ctx context.Context, cfg *transport.Config) error {
```

Comment:
We can't change, transport is an internal package.
```suggestion
func Transport(ctx context.Context, cfg *Config) error {
```